File: presentation/src/main/java/com/dagger/module/AppModule.java
package com.dagger.module;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private final Context mContext;

    public AppModule(Context context) {
        mContext = context;
    }

    @Provides
    @Singleton
    Context provideApplicationContext() {
        return mContext;
    }

}

File: presentation/src/main/java/com/dagger/module/AppModule.kt
package com.dagger.module

import android.content.Context
import com.data.prefs.datasource.PrefsDataSource
import com.data.prefs.repository.SharedPrefsRepositoryImpl
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(context: Context) {
    private val mContext = context

    @Provides
    @Singleton
    fun provideContext(): Context = mContext


    @Provides
    @Singleton
    fun providePrefsDataSource(context: Context): PrefsDataSource = PrefsDataSource(context)

    @Provides
    @Singleton
    fun providePrefsRepository(
            prefsDataSource: PrefsDataSource): SharedPrefsRepositoryImpl = SharedPrefsRepositoryImpl(
            prefsDataSource)

}
